File: modules/agent/src/models/SavedReportProject.py
"""SavedReportProject junction table for many-to-many relationship."""

from sqlalchemy import Column, BigInteger, ForeignKey, DateTime
from sqlalchemy.sql import func

from models import Base


class SavedReportProject(Base):
    """Junction table linking SavedReport to Project (many-to-many)."""

    __tablename__ = "Saved_Report_Project"

    saved_report_id = Column(
        BigInteger,
        ForeignKey("Saved_Report.id", ondelete="CASCADE"),
        primary_key=True
    )
    project_id = Column(
        BigInteger,
        ForeignKey("Project.id", ondelete="CASCADE"),
        primary_key=True
    )
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)

File: modules/agent-go/internal/models/SavedReportProject.go
package models

import "time"

type SavedReportProject struct {
	SavedReportID int64     `gorm:"primaryKey" json:"saved_report_id"`
	ProjectID     int64     `gorm:"primaryKey" json:"project_id"`
	CreatedAt     time.Time `gorm:"autoCreateTime" json:"created_at"`
	UpdatedAt     time.Time `gorm:"autoUpdateTime" json:"updated_at"`
}

func (SavedReportProject) TableName() string {
	return "Saved_Report_Project"
}
